packages/dom-adapters/src/BlockToolAdapter/index.ts

```diff
@@ -171,39 +207,70 @@ export class BlockToolAdapter {
       return;
     }
 
-    const action = event.detail.action;
+    if (isNativeInput) {
+      const currentElement = input as HTMLInputElement | HTMLTextAreaElement;
+      const start = currentElement.selectionStart!;
+      const end = currentElement.selectionEnd!;
+
+      const action = event.detail.action;
+
+      switch (action) {
+        case EventAction.Added: {
+          const text = event.detail.data as string;
+
+          currentElement.value = currentElement.value.slice(0, start) + text + currentElement.value.slice(end);
 
-    const start = rangeIndex[0];
-    const end = rangeIndex[1];
+          currentElement.setSelectionRange(start + text.length, start + text.length);
+          break;
+        }
+        case EventAction.Removed: {
+          if (start === end) {
+            currentElement.value = currentElement.value.slice(0, start - 1) +
```

Comment:
Current value could be assigned to a variable before switch statement and reused across the method